config/src/config/mod.rs

```diff
@@ -287,7 +300,8 @@ impl NodeConfig {
         } else {
             self.validator_network = None;
             if self.full_node_networks.is_empty() {
-                self.full_node_networks.push(NetworkConfig::default());
+                let network_config = NetworkConfig::network_with_id(NetworkId::Public);
```

Comment:
@mgorven is this the public network or is it the internal network for VFN or are we even going to use config-builder at all come launch?

Reply:
note: this is the FN directly on the validator

Reply:
I'm not sure when this code is run, but I'm guessing this is the internal validator-fullnode network on the fullnode side. For launch we need fullnodes to have two networks: the validator-fullnode network, and the public unauthenticated network.

I would prefer not to use `config-builder` in production, but we need a way to handle multiple fullnodes on the validator-fullnode network. If the fullnodes can all use the `fullnode_network` key from Vault with random peer IDs when talking to the validator then that should work. We just need to figure out the validator-side of that to trust that `fullnode_network` key.

Reply:
Let's talk offline about this and come up with a scheme that works. Networking is complicated :/